google_memcache/main.tf

```diff
@@ -0,0 +1,47 @@
+/**
+ * # Terraform Module: Memcache
+ * Creates a memcache instance within GCP using Cloud Memorystore
+ */
+
+locals {
+  default_name             = "${var.application}-${var.realm}-${var.environment}"
+  name                     = coalesce(var.custom_name, local.default_name)
+  default_memcache_configs = {} # Add configs that should be added to all memcache instances here
+  memcache_configs         = merge(local.default_memcache_configs, var.memcache_configs)
+  authorized_network       = trimprefix(var.authorized_network, "https://www.googleapis.com/compute/v1/") # Remove the initial part of the URL if the full network URL is passed.
```

Comment:
@jasonthomas I don't *love* this `trimprefix` part, but unless we add a "short name" to the VPC data object, we may end up doing this for other modules too. The redis module seemed fine taking the whole URL, so I'm not sure how common it will be. I'm fine with keeping this unless you wanted to address it another way.